src/drm.c

```diff
@@ -534,18 +534,14 @@ int modeset_prepare(int fd, struct modeset_output *output_list, uint16_t mode_wi
 
 		out = modeset_output_create(fd, res, conn, mode_width, mode_height, mode_vrefresh);
 		drmModeFreeConnector(conn);
-		if (!out)
-			continue;
-
-		*output_list = *out;
-	}
-	if (!out) {
-		fprintf(stderr, "couldn't create any outputs\n");
-		return -1;
+		if (out) {
+			drmModeFreeResources(res);
+			return out;
```

Comment:
now we return immediately as soon as we detect a monitor connected to HDMI instead of keeping iterating through "connectors". So if the SBC has several HDMI outputs, pixelpilot will choose the first HDMI with connected display.